pyiron/atomistics/structure/atoms.py

```diff
@@ -1118,32 +1118,95 @@ def get_extended_positions(self, width):
         v_repeated = v_repeated[check_dist]
         return v_repeated, indices
 
+    def get_number_of_neighbors_in_sphere(
+            self,
+            cutoff_radius=10,
+            num_neighbors=None,
+            tolerance=2,
+            id_list=None,
+            boundary_width_factor=1.2,
+            num_neighbors_estimate_buffer=1.0,
+    ):
+        """
+        Function to compute the maximum number of neighbors in a sphere around each atom.
+        Args:
+            cutoff_radius (float): Upper bound of the distance to which the search must be done
+            num_neighbors (int/None): maximum number of neighbors found
+            tolerance (int): tolerance (round decimal points) used for computing neighbor shells
+            id_list (list): list of atoms the neighbors are to be looked for
+            boundary_width_factor (float): width of the layer to be added to account for pbc.
+            num_neighbors_estimate_buffer (float): Extra volume taken into account for the num_neighbors estimation
+
+        Returns:
+            maximum number of neighbors found in the sphere of radius cutoff_radius (<= num_neighbors if specified)
+        """
+        if num_neighbors is not None:
+            neigh = self._get_neighbors(
+                num_neighbors=num_neighbors,
+                t_vec=False,
+                tolerance=tolerance,
+                id_list=id_list,
+                cutoff_radius=cutoff_radius,
+                boundary_width_factor=boundary_width_factor,
+            )
+            num_neighbors = np.sum(neigh.distances < np.inf, axis=-1).max()
+        else:
+            volume_per_atom = self.get_volume(per_atom=True)
+            if id_list is not None:
+                volume_per_atom = self.get_volume() / len(id_list)
+            num_neighbors = int((1 + num_neighbors_estimate_buffer) *
+                                4. / 3. * np.pi * cutoff_radius ** 3 / volume_per_atom)
+            num_neighbors_old = 1
+            while num_neighbors_old < num_neighbors:
+                neigh = self._get_neighbors(
+                    num_neighbors=num_neighbors,
+                    t_vec=False,
+                    tolerance=tolerance,
+                    id_list=id_list,
+                    cutoff_radius=cutoff_radius,
+                    boundary_width_factor=boundary_width_factor,
+                )
+                num_neighbors_old = num_neighbors
+                num_neighbors = np.sum(neigh.distances < np.inf, axis=-1).max()
+                if num_neighbors == num_neighbors_old:
+                    num_neighbors = 2 * num_neighbors
+        return num_neighbors
```

Comment:
It's maybe my personal preference, but I might find it very useful if this function returns the number of neighbors for each atom instead of returning the maximum number, because getting the maximum from a numpy array should be easy, since it only requires `.max()`

Reply:
True, and since we do not need it internally any more, this is added value.